### DISCUSSION TITLE
How to use reset with react-select?

### DISCUSSION THREAD

**Discussion by rinatsakaev** [react-hook-form/react-hook-form#3401] [Q&A]
Hello. I have a Select which I want to fill with previously populated form, here in example value is  { value: "vanilla", label: "Vanilla" }.
I use reset() to do this, but when I send this form without changing the Select, handleSubmit returns {select: "[object Object]"}.
In case if I change Select, everything is ok because onChange is fired.
How to send a form correctly if I don't modify Select?
You can check it here:
https://codesandbox.io/s/react-hook-form-v6-customise-controller-return-value-forked-d5z6k?file=/src/index.js
press submit and check console.

**Comment by bluebill1049** [comment]
https://github.com/react-hook-form/react-hook-form/tree/master/examples

search: Controller modify return value

**Reply by rinatsakaev** [reply]
In your example If I add reset, value and label would be the same. Also, it's not possible to access label in render function (line 41).
https://codesandbox.io/s/react-hook-form-v6-customise-controller-return-value-forked-6dlni?file=/src/index.js

**Reply by bluebill1049** [reply]
why you need to have read label in the render, if you want to watch that value use `watch` api.

**Reply by rinatsakaev** [reply]
I just want Select to render label, not value

**Reply by bluebill1049** [reply]
`onChange(e.label);`?

**Reply by rinatsakaev** [reply]
No, if I would use onChange(e.label) then handleSubmit will return label, not value. Problem is that in [this example](https://codesandbox.io/s/react-hook-form-v6-customise-controller-return-value-forked-6dlni?file=/src/index.js) Select shows 'myValue' instead of 'myLabel'. This is beacause of `           
 value={{
              value: value,
              label: value
            }}`. But I can't write `            value={{
              value: value.value,
              label: value.label
            }}` because 

1. In reset() I specify only value
2. In onChange I specify only value
3. Argument 'value' of render() has what I specify in onChange which is value.
![image](https://user-images.githubusercontent.com/24736895/98642921-0bf69f80-2350-11eb-9c91-93e31b2369c8.png)

**Reply by anandagarwal6040** [reply]
I am facing similar issue did any body find any solution?

**Reply by Moshyfawn** [reply]
> I am facing similar issue did any body find any solution?

Please, provide a minimal csb with the version of RHF your're facing the issue with

**Comment by Froidoh** [comment]
One problem I have no solution for:

When I do not send the form but reset it, everything works like a charm and the `react-select` value is reset to the `defaultValue`.

If however I send the form and apply the result (so I get a response back with the "form-entity" as it is after the update) all the fields are set correctly.

If I change something and reset, TextInputs and so on are set to the newest version (the one after the update), but the react-select value is set to the initial version it had before the update.

Like:

Open a form... name: "Nemo", country: "Australia"

-> name: "Wanda", country: "USA"
-> update()

<- name: "Wanda", country: "USA"

reset()

name: "Wanda", country: "Australia" (!!! error here, Country should be USA, it was already saved!)

**Reply by leapful** [reply]
@Froidoh 

Please provide a codesandbox to re-produce the issue.
